Migration: New porting guide from Cloud team

theotherjimmy opened the pull request:
Original text from @jen3andruska:
> New porting guide for porting Mbed OS before porting Mbed Cloud

theotherjimmy commented:
I can at least rebase/filter-branch out the directory capitalization for you

theotherjimmy commented:
Got it. Now the dir names match convention on this repo.

ashok-rao commented:
Thanks @theotherjimmy ..
@iriark01 @AnotherButler : How do we proceed with this now?

theotherjimmy commented:
Alright, I have updated the engineering content, removed passive vioce where possible and re-did the formatting to match other documents in this repo. Now: how can we merge this with the other tutorial in this repo.

iriark01 commented:
I suggest we wait for Amanda. She's back on Monday.

Thank you for all your help!

ashok-rao commented:
@AnotherButler .. ping (hoping you are back in office)!

AnotherButler commented:
@ashok-rao I'm reviewing this, and then I may move it into our porting section. I hope to have my edits complete tomorrow morning my time.

AnotherButler commented on docs/tutorials/porting.md:
```diff
@@ -0,0 +1,225 @@
+## Mbed OS porting guide
```
I don't see any of the images on this page. Do we want to include them here or delete them?

theotherjimmy replied:
The last commit replaced all of them with the single tree structure at the bottom of the document.

AnotherButler commented on docs/tutorials/porting.md:
```diff
@@ -0,0 +1,225 @@
+- Requires the macro `CPU_DEVICE_1` to compile the vendor-provided HAL for the device mounted on the board.
+- Requires the macro `FAMILY_MY_FAMILY` to compile the vendor-provided HAL for the family of the device.
+
+All of these requirements directly map to relevant tags in the `targets.json` entry for our target. This is shown in step 3 below.
```
Note: We don't have a step 3 anymore. What does this reference?

theotherjimmy replied:
The following step.

theotherjimmy commented:
@AnotherButler I think we can remove the "Amanda is on holiday. please hold." label (which I incidentally find funny).

iriark01 commented:
Oh I put that there...

AnotherButler commented:
Closing in favor of #625
